### PR TITLE
Adding option to disable environment name updates when in debug mode

### PR DESCRIPTION
**Pull request by fabiocav**
### Issue describing the changes in this PR

This addresses a common source for #6239

Changing this behavior would be a breaking change. Adding this behind a feature flag for now (@brettsam , this is something I want to enable by default in 4.0)

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: release_notes.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <!-- Please add your release notes in the following format:
 - My change description (#PR)
 -->
+- Added a feature flag to opt out of the default behavior where the host sets the environment name to `Development` when running in debug mode. To disable the behavior, set the app setting: `AzureWebJobsFeatureFlags` to `DisableDevModeInDebug`
```
Comment (brettsam) [inline release_notes.md:5]: Should the setting here specifically mention the Dev *Environment*? Don't other places check `InDebugMode` as well? Or is that it?

Comment (fabiocav) [reply release_notes.md:5]: Other places use the debug mode flag to apply logic to logging, but that's not a dev mode. I'm open to other suggestions for this name.

Comment (brettsam) [reply release_notes.md:5]: I also think there's code that changes things like the lease expirations and queue polling delay, etc...

Comment (fabiocav) [reply release_notes.md:5]: Yes, that applies to the debug and local dev as well.

**Review by ankitkumarr — APPROVED** [review]

**Pull request merged**